Classes/Utility/GeneralUtility.php

```diff
@@ -365,6 +365,30 @@ public static function doRedirectBasedOnSettings(array $settings, array $gp, str
     self::debugMessage('No redirectPage set.');
   }
 
+  /**
+   * @param array<string, mixed> $conditionSettings
+   * @param array<string, mixed> $gp
+   */
+  public static function evaluateConditions(array $conditionSettings, array $gp): false {
```

Comment:
Change to private static and return type must be bool.  

The whole point is that we start with  `$evaluation = false;` and `eval('$evaluation = '.$finalCondition.';');` could change it to true.

Reply:
With the new evaluateConditions function I ok with leaving it public.